### PR TITLE
Remove protobufs from Android

### PR DESCRIPTION
**Pull request by outlandnish**
- Updated the Android library to use HashMaps instead of protobufs
- Probably could use some updates to the naming of variables. This PR was a bit quick and dirty to get to a functioning state
- There seems to be an issue with the initial request of MTU in the example. Is this because of a change in the way the plugin should be called?

### CODE REVIEW

**Comment by chipweinberger** [comment]
awesome work!!!!

let me know when you think it is ready to merge!

**Comment by outlandnish** [comment]
It was mostly working, but I think I had some bugs in my app - I'll double check it this week and shoot you a message!

File: android/src/main/java/com/boskokg/flutter_blue_plus/MessageMaker.java [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+        for(BluetoothGattService s : service.getIncludedServices()) {
+            includedServices.add(from(device, s, gatt));
+        }
+        dev.put("included_services", includedServices);
+        return dev;
+    }
+
+    static HashMap<String, Object> from(BluetoothDevice device, BluetoothGattCharacteristic characteristic, BluetoothGatt gatt) {
+        HashMap<String, Object> ch = new HashMap<>();
+        ch.put("remoteId", device.getAddress());
```
Comment (outlandnish) [inline android/src/main/java/com/boskokg/flutter_blue_plus/MessageMaker.java:139]: Need to change this from `remoteId` to `remote_id`

**Comment by chipweinberger** [comment]
Also it looks like you reformatted the dart code. 

For the purpose of a simple diff, we should remove that.

**Comment by outlandnish** [comment]
> Also it looks like you reformatted the dart code.
> 
> For the purpose of a simple diff, we should remove that.

Ah sorry about that - I think I have it auto re-formatting in VS Code on save so that happened. I'll undo that change.

**Comment by chipweinberger** [comment]
Let me know when it should be merged. thanks.

**Comment by outlandnish** [comment]
@chipweinberger I'm not sure I can easily undo the formatting changes, but the library does now work! Let me know if it functionally does work and then we can dig into how to get the formatting back to the way it should be.

**Comment by chipweinberger** [comment]
what's the status on this PR? :)

**Comment by chipweinberger** [comment]
I merged most of the changes here: https://github.com/boskokg/flutter_blue_plus/commit/8a56e6959bc6316041ef68f887976a484a9aec8d

Feel free to open a PR for some of the other changes.

**Comment by outlandnish** [comment]
> what's the status on this PR? :)

Sorry, I've been on vacation for the last week! I'll be getting back to it this evening to see what else still needs to be updated to make it work. I've noticed a couple of issues in testing - have you been able to try it yourself?

**Comment by Staars** [comment]
I did a test and it is not working for me. Bluetooth is unavailable.
Debug output sample:
```
/DartMessenger(28821): Uncaught exception in binary message listener
E/DartMessenger(28821): java.lang.IllegalStateException: Reply already submitted
E/DartMessenger(28821): 	at io.flutter.embedding.engine.dart.DartMessenger$Reply.reply(DartMessenger.java:435)
E/DartMessenger(28821): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:278)
E/DartMessenger(28821): 	at io.flutter.embedding.engine.dart.DartMessenger.invokeHandler(DartMessenger.java:295)
E/DartMessenger(28821): 	at io.flutter.embedding.engine.dart.DartMessenger.lambda$dispatchMessageToQueue$0$io-flutter-embedding-engine-dart-DartMessenger(DartMessenger.java:322)
E/DartMessenger(28821): 	at io.flutter.embedding.engine.dart.DartMessenger$$ExternalSyntheticLambda0.run(D8$$SyntheticClass)
E/DartMessenger(28821): 	at android.os.Handler.handleCallback(Handler.java:836)
E/DartMessenger(28821): 	at android.os.Handler.dispatchMessage(Handler.java:103)
E/DartMessenger(28821): 	at android.os.Looper.loop(Looper.java:203)
E/DartMessenger(28821): 	at android.app.ActivityThread.main(ActivityThread.java:6251)
E/DartMessenger(28821): 	at java.lang.reflect.Method.invoke(Native Method)
E/DartMessenger(28821): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1063)
E/DartMessenger(28821): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:924)
```

Switching back to the last working Android version for me with (in pub spec.yaml):
```
  flutter_blue_plus:
      git: 
        url: https://github.com/boskokg/flutter_blue_plus.git
        ref: 1.6.1 # use for Android, comment out for macOS
```
brought back Bluetooth immediately.

#361

**Comment by chipweinberger** [comment]
I see in a few places we are forgetting to call `break`.

```
                if (mBluetoothAdapter.isEnabled()) {
                    result.success(true); // no work to do
                }
```

**Comment by chipweinberger** [comment]
merged https://github.com/boskokg/flutter_blue_plus/commit/bc8c03c9a8b7ab1c975ae73a845a9fea7c1c0f14

**Comment by JDongKhan** [comment]
@chipweinberger  all case need break; 
屏幕录制2023-07-08 10 35 22

**Comment by JDongKhan** [comment]
https://github.com/boskokg/flutter_blue_plus/assets/4929805/cb8d8d47-5ed1-4cd7-b4a4-cec511a1d76a

**Comment by chipweinberger** [comment]
good point.

**Comment by chipweinberger** [comment]
master branch should now be at a decent state. needs testing tho.

please rebase any future changes on top of master.

File: example/android/app/build.gradle [hunk 2]
```diff
@@ -22,10 +22,13 @@ if (flutterVersionName == null) {
 }
 
 apply plugin: 'com.android.application'
+apply plugin: 'kotlin-android'
 apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
 
 android {
+    namespace "com.boskokg.flutter_blue_plus_example"
     compileSdkVersion 33
+    ndkVersion flutter.ndkVersion
```
Comment (MrCsabaToth) [inline example/android/app/build.gradle:31]: Do we NDK?

**Comment by chipweinberger** [comment]
closing since these changes have been mostly merged

**Pull request closed without merging**